src/js/components/Button/StyledButton.js

```diff
@@ -11,18 +11,45 @@ import {
 } from '../../utils';
 import { defaultProps } from '../../default-props';
 
+const radiusStyle = props => {
+  if (props.size && props.theme.button.size) {
+    return props.theme.button.size[props.size].border.radius;
+  }
+  return props.theme.button.border.radius;
+};
+
+const fontStyle = props => {
+  const size = props.size || 'medium';
+  const data = props.theme.text[size];
+  return css`
+    font-size: ${data.size};
+    line-height: ${data.height};
+  `;
+};
+
+const padStyle = props => {
+  if (props.size && props.theme.button.size) {
```

Comment:
Similar to radius comment above, I think we should check for `props.theme.button.size[props.size]` too.